### ISSUE TITLE
Get peer-invites into Patchbay

### ISSUE THREAD

**Issue by mixmix** [ssbc/patchbay#341]
_This is an issue I would like to work on as a Gitcoin bounty. If you're coming from Gitcoin, please don't spend time applying to work on this._

---

The current system of invite codes forces many people to jam into the same pub. This causes social / safety issues.

dominictarr has made `ssb-peer-invites` which will help connect peers more directly, with pubs as even more demoted in their role, making them more like introducers, which has been the goal for a long time.

## Solution 

A lot of the back end work has been done, but we need to work on interfaces which make this accessible, so people can generate peer-based invites, and others can accept them.

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__This issue now has a funding of 500.0 DAI (500.0 USD @ $1.0/DAI)  attached to it.__

 * If you would like to work on this issue you can 'start work' [on the Gitcoin Issue Details page](https://gitcoin.co/issue/ssbc/patchbay/341/3148).
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/ssbc/patchbay/341/3148).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
* $86,069.99 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__Workers have applied to start work__.


These users each claimed they can complete the work by 12 months from now.
Please review their action plans below:


**1) [mixmix](https://gitcoin.co/profile/mixmix) has applied to start work _(Funders only: [approve worker](https://gitcoin.co/issue/ssbc/patchbay/341/3148?mutate_worker_action=approve&worker=mixmix) | [reject worker](https://gitcoin.co/issue/ssbc/patchbay/341/3148?mutate_worker_action=reject&worker=mixmix))_.**

I'm going to add a similar field to the /network page as we have for old-school invites.
I will also do some work to add peer-invites as something you can use in the initial onboarding tools offered (by ssb-ahoy) when you first launch patchbay,

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/ssbc/patchbay/341/3148).

**Comment by arj03** [comment]
wop wop wop 🚀

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__Workers have applied to start work__.


These users each claimed they can complete the work by 12 months from now.
Please review their action plans below:


**1) [mixmix](https://gitcoin.co/profile/mixmix) has applied to start work _(Funders only: [approve worker](https://gitcoin.co/issue/ssbc/patchbay/341/3148?mutate_worker_action=approve&worker=mixmix) | [reject worker](https://gitcoin.co/issue/ssbc/patchbay/341/3148?mutate_worker_action=reject&worker=mixmix))_.**

I'm going to add a similar field to the /network page as we have for old-school invites.
I will also do some work to add peer-invites as something you can use in the initial onboarding tools offered (by ssb-ahoy) when you first launch patchbay,
**2) [sanchaymittal](https://gitcoin.co/profile/sanchaymittal) has applied to start work _(Funders only: [approve worker](https://gitcoin.co/issue/ssbc/patchbay/341/3148?mutate_worker_action=approve&worker=sanchaymittal) | [reject worker](https://gitcoin.co/issue/ssbc/patchbay/341/3148?mutate_worker_action=reject&worker=sanchaymittal))_.**

I will integrate the back end work to the button on the front end at the network page and set the threshold of authentication(if required). Update to the front end relative to the backend work to make it more interactive and user-friendly.

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/ssbc/patchbay/341/3148).

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. **Started** 3. Submitted 4. Done 

<hr>

__Work has been started__.


These users each claimed they can complete the work by 12 months from now.
Please review their action plans below:


**1) [mixmix](https://gitcoin.co/profile/mixmix) has been approved to start work.**

I'm going to add a similar field to the /network page as we have for old-school invites.
I will also do some work to add peer-invites as something you can use in the initial onboarding tools offered (by ssb-ahoy) when you first launch patchbay,
**2) [sanchaymittal](https://gitcoin.co/profile/sanchaymittal) has applied to start work _(Funders only: [approve worker](https://gitcoin.co/issue/ssbc/patchbay/341/3148?mutate_worker_action=approve&worker=sanchaymittal) | [reject worker](https://gitcoin.co/issue/ssbc/patchbay/341/3148?mutate_worker_action=reject&worker=sanchaymittal))_.**

I will integrate the back end work to the button on the front end at the network page and set the threshold of authentication(if required). Update to the front end relative to the backend work to make it more interactive and user-friendly.

Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/ssbc/patchbay/341/3148).

**Comment by mixmix** [comment]
hooray. It's the weekend now, but I'm going to start into this after I've had a rest

**Comment by mixmix** [comment]
getting started on this now!

**Comment by gitcoinbot** [comment]
@mixmix Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] reminder (3 days)
* [ ] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/ssbc/patchbay/341/3148?snooze=1) | [3 days](https://gitcoin.co/issue/ssbc/patchbay/341/3148?snooze=3) | [5 days](https://gitcoin.co/issue/ssbc/patchbay/341/3148?snooze=5) | [10 days](https://gitcoin.co/issue/ssbc/patchbay/341/3148?snooze=10) | [100 days](https://gitcoin.co/issue/ssbc/patchbay/341/3148?snooze=100)

**Comment by mixmix** [comment]
I'm working on this now.
I'll open a WIP PR today

**Comment by mixmix** [comment]
WIP PR - https://github.com/ssbc/patchbay/pull/349

**Comment by gitcoinbot** [comment]
@mixmix Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] reminder (3 days)
* [ ] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/ssbc/patchbay/341/3148?snooze=1) | [3 days](https://gitcoin.co/issue/ssbc/patchbay/341/3148?snooze=3) | [5 days](https://gitcoin.co/issue/ssbc/patchbay/341/3148?snooze=5) | [10 days](https://gitcoin.co/issue/ssbc/patchbay/341/3148?snooze=10) | [100 days](https://gitcoin.co/issue/ssbc/patchbay/341/3148?snooze=100)

**Comment by gitcoinbot** [comment]
@mixmix Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] reminder (3 days)
* [ ] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/ssbc/patchbay/341/3148?snooze=1) | [3 days](https://gitcoin.co/issue/ssbc/patchbay/341/3148?snooze=3) | [5 days](https://gitcoin.co/issue/ssbc/patchbay/341/3148?snooze=5) | [10 days](https://gitcoin.co/issue/ssbc/patchbay/341/3148?snooze=10) | [100 days](https://gitcoin.co/issue/ssbc/patchbay/341/3148?snooze=100)

**Comment by mixmix** [comment]
Yes still working on this. You'll see I pushed commits yesterday.
Solving some lower level problems which are blocking me

**Comment by gitcoinbot** [comment]
@mixmix Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] reminder (3 days)
* [ ] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/ssbc/patchbay/341/3148?snooze=1) | [3 days](https://gitcoin.co/issue/ssbc/patchbay/341/3148?snooze=3) | [5 days](https://gitcoin.co/issue/ssbc/patchbay/341/3148?snooze=5) | [10 days](https://gitcoin.co/issue/ssbc/patchbay/341/3148?snooze=10) | [100 days](https://gitcoin.co/issue/ssbc/patchbay/341/3148?snooze=100)

**Comment by mixmix** [comment]
Still working on this. Waiting on another PR I had to open on
ssb-peer-invites

**Comment by gitcoinbot** [comment]
@mixmix Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] reminder (3 days)
* [ ] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/ssbc/patchbay/341/3148?snooze=1) | [3 days](https://gitcoin.co/issue/ssbc/patchbay/341/3148?snooze=3) | [5 days](https://gitcoin.co/issue/ssbc/patchbay/341/3148?snooze=5) | [10 days](https://gitcoin.co/issue/ssbc/patchbay/341/3148?snooze=10) | [100 days](https://gitcoin.co/issue/ssbc/patchbay/341/3148?snooze=100)

**Comment by mixmix** [comment]
Still going

**Comment by mixmix** [comment]
Here's a link to the work I've been getting done lower level for reference : https://github.com/ssbc/ssb-peer-invites/pull/22

**Comment by mixmix** [comment]
This work was merged here : https://github.com/ssbc/patchbay/pull/349

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. Started 3. **Submitted** 4. Done 

<hr>

__Work for 500.0 DAI (500.0 USD @ $1.0/DAI) has been submitted by__:

 1. [@mixmix](https://gitcoin.co/profile/mixmix)



@ceresstation please take a look at the submitted work:
* [PR](https://github.com/ssbc/patchbay/pull/349) by @mixmix

<hr>

* Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/ssbc/patchbay/341/3148)
* Want to chip in? Add your own contribution [here](https://gitcoin.co/issue/ssbc/patchbay/341/3148).
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
* $163,462.08 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. Started 3. Submitted 4. **Done** 

<hr>

__The funding of 500.0 DAI (500.0 USD @ $1.0/DAI)  attached to this issue has been approved & issued to @mixmix.__  

 
* Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/ssbc/patchbay/341/3148)
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
* $161,802.51 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)
